### PR TITLE
Update Nyx harness

### PR DESCRIPTION
**Pull request by jschwartzentruber**

### CODE REVIEW

File: src/guided_fuzzing_daemon/nyx.py [hunk 1]
```diff
@@ -63,7 +63,7 @@ def update_and_write(
         any_stat = False
 
         def convert_num(num: str) -> Union[float, int]:
```
Comment (tysmith) [inline src/guided_fuzzing_daemon/nyx.py:66]: Why not always just return a float?

File: src/guided_fuzzing_daemon/args.py [hunk 2]
```diff
@@ -1,23 +1,40 @@
...
 from typing import List, Optional
 
 from .utils import HAVE_FFPUPPET
 
 
+def _check_log_pattern(
+    instances: int, pattern: str, arg_name: str, parser: ArgumentParser
+) -> None:
+    if instances > 1 and pattern.count("%") != 1:
+        parser.error(f"{arg_name} expects exactly one %d pattern")
```
Comment (tysmith) [inline src/guided_fuzzing_daemon/args.py:17]: Stray %d

**Review by tysmith — CHANGES_REQUESTED** [review]
I think this should have better test coverage before we merge it.

File: src/guided_fuzzing_daemon/nyx.py (hunk 1 above)
Comment (jschwartzentruber) [reply src/guided_fuzzing_daemon/nyx.py:64]: This preserves the AFL output format (int stays int in the stats output) without hardcoding which is which.

File: src/guided_fuzzing_daemon/args.py (hunk 2 above)
Comment (jschwartzentruber) [reply src/guided_fuzzing_daemon/args.py]: This is intended to be %d.

**Review by tysmith — APPROVED** [review]

**Pull request merged**